src/Features/Lsif/Generator/ResultSetTracking/SymbolHoldingResultSetTracker.cs

```diff
@@ -72,56 +72,22 @@ private TrackedResultSet GetTrackedResultSet(ISymbol symbol)
             //
             // This we do outside the lock -- whichever thread was the one to create this was the one that
             // gets to write out the moniker, but others can use the ResultSet Id at this point.
-            if (symbol.OriginalDefinition.Equals(symbol))
+            if (SymbolMoniker.HasMoniker(symbol))
             {
-                var monikerVertex = TryCreateMonikerVertexForSymbol(symbol);
-
-                if (monikerVertex != null)
-                {
-                    // Attach the moniker vertex for this result set
-                    _ = GetResultIdForSymbol(symbol, "moniker", () => monikerVertex);
-                }
+                _ = this.GetMoniker(symbol, _sourceCompilation);
```

Comment:
i don't really understand this, but ok :)

Reply:
this work doesn't seem complex, is it actually bad to do under the lock?

Reply:
The lock earlier is a pretty high contention lock -- we have to hit it to look up a symbol -> id mapping every time we process any token, so it seems best to keep it out.

This change makes things a bit more robust anyways -- the other place assuming there was a moniker might also be wrong if for some reason we didn't have a moniker for that symbol so there's no real concern about the race anymore.